### ISSUE TITLE
Canary test rgw-multisite-testing is failing since Oct 3rd

### ISSUE THREAD

**Issue by travisn** [rook/rook#14841]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The `rgw-multisite-testing` canary test is failing consistently since Oct 3rd. There were two PRs merged that day related to object store (#14772 and #14801), as well as a large change to remove support for Quincy (#14795). 

@jhoblitt Could you take a look to see if it could be related to your changes? It's not obvious to me where the issues lies.

An example failure from the [daily CI](https://github.com/rook/rook/actions/workflows/daily-nightly-jobs.yml) is shown here:
```
  tests/scripts/github-action-helper.sh test_multisite_object_replication
  shell: /usr/bin/bash --noprofile --norc -eo pipefail -x {0}
+ tests/scripts/github-action-helper.sh test_multisite_object_replication
++ readlink -f -- tests/scripts/../..
+ REPO_DIR=/home/runner/work/rook/rook
+ NETWORK_ERROR='connection reset by peer'
+ SERVICE_UNAVAILABLE_ERROR='Service Unavailable'
+ INTERNAL_ERROR=INTERNAL_ERROR
+ INTERNAL_SERVER_ERROR='500 Internal Server Error'
+ FUNCTION=test_multisite_object_replication
+ shift
+ test_multisite_object_replication
++ get_secret_key rook-ceph realm-a-keys access-key
++ local ns=rook-ceph
++ local secret_name=realm-a-keys
++ local key=access-key
++ base64 --decode
++ kubectl -n rook-ceph get secrets realm-a-keys -o 'jsonpath={.data.access-key}'
+ S3CMD_ACCESS_KEY=JDl6Z0d1LEUrJytBKEc=
+ readonly S3CMD_ACCESS_KEY
++ get_secret_key rook-ceph realm-a-keys secret-key
++ local ns=rook-ceph
++ local secret_name=realm-a-keys
++ local key=secret-key
++ base64 --decode
++ kubectl -n rook-ceph get secrets realm-a-keys -o 'jsonpath={.data.secret-key}'
+ S3CMD_SECRET_KEY=Zk5cQixUVFpKLid6I0NTNDRFIkJ6QDcxTSExRA==
+ readonly S3CMD_SECRET_KEY
+ local cluster_1_ip
++ get_clusterip rook-ceph rook-ceph-rgw-multisite-store
++ local ns=rook-ceph
++ local cluster_name=rook-ceph-rgw-multisite-store
++ kubectl -n rook-ceph get svc rook-ceph-rgw-multisite-store -o 'jsonpath={.spec.clusterIP}'
+ cluster_1_ip=10.101.21.87
+ local cluster_2_ip
++ get_clusterip rook-ceph-secondary rook-ceph-rgw-zone-b-multisite-store
++ local ns=rook-ceph-secondary
++ local cluster_name=rook-ceph-rgw-zone-b-multisite-store
++ kubectl -n rook-ceph-secondary get svc rook-ceph-rgw-zone-b-multisite-store -o 'jsonpath={.spec.clusterIP}'
+ cluster_2_ip=10.99.56.124
+ cd /home/runner/work/rook/rook/deploy/examples
+ cat
+ write_object_read_from_replica_cluster 10.101.21.87 10.99.56.124 test1
+ local write_cluster_ip=10.101.21.87
+ local read_cluster_ip=10.99.56.124
+ local test_bucket_name=test1
+ local test_object_name=test1-1mib-test.dat
+ fallocate -l 1M test1-1mib-test.dat
+ echo test1-1mib-test.dat
+ s3cmd --host=10.101.21.87 mb s3://test1
+ command timeout 20 s3cmd -v --config=s3cfg --access_key=JDl6Z0d1LEUrJytBKEc= --secret_key=Zk5cQixUVFpKLid6I0NTNDRFIkJ6QDcxTSExRA== --host=10.101.21.87 mb s3://test1

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    An unexpected error has occurred.
  Please try reproducing the error using
  the latest s3cmd code from the git master
  branch found at:
    https://github.com/s3tools/s3cmd
  and have a look at the known issues list:
    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)
  If the error persists, please report the
  following lines (removing any private
  info as necessary) to:
   s3tools-bugs@lists.sourceforge.net


!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Invoked as: /usr/bin/s3cmd -v --config=s3cfg --access_key=JDl6Z0d1LEUrJytBKEc= --secret_key=Zk5cQixUVFpKLid6I0NTNDRFIkJ6QDcxTSExRA== --host=10.101.21.87 mb s3://test1
Problem: <class 'ConnectionRefusedError: [Errno 111] Connection refused
S3cmd:   2.2.0
python:   3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0]
environment LANG=C.UTF-8

Traceback (most recent call last):
  File "/usr/bin/s3cmd", line 3209, in <module>
    rc = main()
  File "/usr/bin/s3cmd", line 3106, in main
    rc = cmd_func(args)
  File "/usr/bin/s3cmd", line 260, in cmd_bucket_create
    response = s3.bucket_create(uri.bucket(), cfg.bucket_location, cfg.extra_headers)
  File "/usr/lib/python3/dist-packages/S3/S3.py", line 430, in bucket_create
    response = self.send_request(request)
  File "/usr/lib/python3/dist-packages/S3/S3.py", line 1480, in send_request
    conn = ConnMan.get(self.get_hostname(resource['bucket']))
  File "/usr/lib/python3/dist-packages/S3/ConnMan.py", line 284, in get
    conn.c.connect()
  File "/usr/lib/python3.10/http/client.py", line 942, in connect
    self.sock = self._create_connection(
  File "/usr/lib/python3.10/socket.py", line 845, in create_connection
    raise err
  File "/usr/lib/python3.10/socket.py", line 833, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    An unexpected error has occurred.
  Please try reproducing the error using
  the latest s3cmd code from the git master
  branch found at:
    https://github.com/s3tools/s3cmd
  and have a look at the known issues list:
    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions-(FAQ)
  If the error persists, please report the
  above lines (removing any private
  info as necessary) to:
   s3tools-bugs@lists.sourceforge.net
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Error: Process completed with exit code 74.
```

**Expected behavior:**
Integration tests should pass.

**How to reproduce it (minimal and precise):**

Run the CI for a PR or the daily CI.

**Comment by jhoblitt** [comment]
The reliability issue predates Oct 3rd.  I was manually re-triggering that job on Oct 1st: https://github.com/rook/rook/actions/runs/11136011606/job/30951918760?pr=14789

**Comment by jhoblitt** [comment]
The existing logic wasn't doing what I thought it was... it is checking for the rgw pods to be ready rather than the `CephObjectStore` resource to be ready. I don't know if that is "the" problem but it is likely "a" problem.

**Comment by jhoblitt** [comment]
After changing the logic to wait for the rook kinds to have a ready status, instead of waiting on pods, and increasing some timeouts, I've gotten CI to pass 3 times in a row.  It is possible that is a fluke but I'm hopefully that #14842 has at least reduced the failure rate.

**Comment by travisn** [comment]
Yes, looks great to go ahead and merge that and go from there!

**Comment by jhoblitt** [comment]
The plot thickens.  https://github.com/rook/rook/pull/14793 is seeing issues with the `two-object-one-zone` canary job where the cephobjectzone/cephobjectstore CRs are not transitioning to Ready.  Perhaps `cephobjectzone` is the common factor?  https://github.com/rook/rook/actions/runs/11333158606/job/31516776467?pr=14793

**Comment by jhoblitt** [comment]
Or maybe not.  The zone issue was self inflicted. 

```
  Warning  ReconcileFailed  9m                     rook-ceph-object-zone-controller  failed to reconcile CephObjectZone "rook-ceph/two-object-one-zone". ceph zone group "two-object-one-zone"
 not found: exit status 2
  Warning  ReconcileFailed  3m6s (x17 over 8m48s)  rook-ceph-object-zone-controller  failed to reconcile CephObjectZone "rook-ceph/two-object-one-zone". failed to create ceph zone: unable to
 create pools for zone: CR store pools are missing: [two-object-one-zone.rgw.control two-object-one-zone.rgw.meta two-object-one-zone.rgw.log two-object-one-zone.rgw.buckets.index two-object
-one-zone.rgw.buckets.non-ec two-object-one-zone.rgw.otp two-object-one-zone.rgw.buckets.data]
```

**Comment by BlaineEXE** [comment]
Fixed in #8785
